R/pin-read-write.R

```diff
@@ -252,39 +252,41 @@ hash_file <- function(path) {
   digest::digest(file = path, algo = "xxhash64")
 }
 
-check_board <- function(x, v1, v0) {
+check_board <- function(x, v1, v0, arg = caller_arg(x), call = caller_env()) {
   if (!inherits(x, "pins_board")) {
-    abort("`board` must be a pin board")
+    stop_input_type(x, "a pin board", arg = arg, call = call)
   }
-
   if (!1 %in% x$api) {
-    this_not_that(v0, v1)
+    this_not_that(v0, v1, call = call)
   }
 }
-check_name <- function(x) {
+
+check_pin_name <- function(x, call = caller_env()) {
   if (grepl("\\\\|/", x, perl = TRUE)) {
-    abort("`name` must not contain slashes", class = "pins_check_name")
+    cli_abort(
+      "{.var name} must not contain slashes",
+      class = "pins_check_name",
```

Comment:
Very minor - We have somewhat gotten in the habit of classing our errors with

```
{pkg}_{error}_{descriptor}

# like
pins_error_invalid_name
```

in practice it doesn't matter much, just letting you know

Reply:
Thanks! This particular one has been around for a while (it predates my work on pins) so I probably won't change it but I'll keep in mind moving forward. 👍